spring-integration-core/src/main/java/org/springframework/integration/config/annotation/MessagingAnnotationPostProcessor.java

```diff
@@ -98,10 +97,10 @@ public void setBeanFactory(BeanFactory beanFactory) {
 	 * @param requireComponentAnnotation the {@code boolean} flag to indicate requirements for the
 	 * {@link Component} annotation presentation for the messaging annotations.
 	 * @since 4.3
-	 * @see org.springframework.integration.context.IntegrationProperties#REQUIRE_COMPONENT_ANNOTATION
+	 * @deprecated since 5.0 with no-op.
```

Comment:
I think it would be better to fail fast here (don't deprecate, just remove). This setter is unlikely to be called by user code and, if it is, I think it would be better (s)he knows it no longer exists.